tests/python/test_slowmo_fsdp.py

```diff
@@ -0,0 +1,382 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+import os
+import sys
+import tempfile
+from typing import Optional
+
+# In case `tsan` checks are enabled, need to skip.
+# Refer to issue: https://github.com/pytorch/torchdistx/issues/39
+if os.getenv("LD_PRELOAD") and "libtsan.so.0" in os.getenv("LD_PRELOAD"):
+    print("Current tests do not work with Thread Sanitizer checks", file=sys.stderr)
+    sys.exit(0)
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import distributed as dist
+from torch.distributed.algorithms.model_averaging import averagers
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp.fully_sharded_data_parallel import ShardingStrategy
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+)
+
+from torchdistx.slow_momentum import slow_momentum_comm, slow_momentum_optimizer
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+
+class Net(nn.Module):
+    def __init__(self, has_wrapping, sharding_strategy):
+        # to ensure determinism
+        torch.manual_seed(0)
+        torch.cuda.manual_seed(0)
+        super().__init__()
+
```

Comment:
nit: For readability maybe define the non-wrapped model first so that it is clear that the base model is the same for both. Then check `if has_wrapping` and in that block, you wrap each submodule with `FSDP`. Plus, you may refactor the `fsdp_kwargs` out since you are always passing the same kwargs to `FSDP` (namely, `device_id` and `sharding_strategy`).